### ISSUE TITLE
release 2019-04 follow-up

### ISSUE THREAD

**Issue by stonebig** [winpython/winpython#776]
release date: December 2019

Expected:
- Tensorflow_cpu-2.1.0rc1, PyTorch-1.3.1
- numpy-1.17
- Python-3.8.1, Python3.7.6, python-vscode 2019.11,
- only ".dot" 32 bit versions

Not Expected:
- Spyder-4 transition is postponed to Spyder-4.1 (issue with in-editor completion due to web port usage)

Focus of the release:
- vscode refined integration :
  - [x] : october update ==>   .ipynb direct opening (but lot of issues, to moderate your enthousiasm see here https://github.com/microsoft/vscode-python/issues?q=is%3Aissue+is%3Aopen+label%3A%22data+science%22)
- push towards micro-services and ASGI:
   - [x] better flask REST tooling
   - [x] ASGI : Quart, uvicorn, FastAPI, hypercorn
- simplify further the build process, to be able to move to the cloud:
    - [x] split in 3 steps : infrastructure creation, packages additions, wrap-up,
    - cleanup & simplify each step:
      -  [x] share Tools directories between Python Versions
      -  [x] share Docs directories between Python Versions
    - [ ] reduce need for an initial full WinPython environnment.

Non-Focus of the release:
- Jupyterlab-2 (far away)
- Spyder-4.0  (works less nice with Windows than previous version, until Spyder-4.1)

Hopes/Wishes for coming 2020:
- Spyder-5 compatible with PySide2  Q2 2020
- nodejs (or webpack or npm ?) no longer necessary to update Jupyterlab-2 Q2 2020,
- Tensorflow-2 and Pytorch get smaller, using windows ONNX infrastructure Q3 2020,
- Python-3.9 hugely simplifies Asyncio (inspired per Trio) and multi-core (sub-interpreters) Q4 2020
- Python hit 32% on PYPL.

**Comment by stonebig** [comment]
audit-trail

**Comment by stonebig** [comment]
build 0 Python-3.8.0rc1

**Comment by stonebig** [comment]
beta1 (plan)

Changes from WinPython 2019-03:
- Infrastructure: 
   - Pythons-3.8.0, Python-3.7.5
- New Packages: 
   - FastAPI, uvicorn, Quart
   - datasette
- Upgraded Packages: 
- variation per version, for build1:
  -  Spyder-3 is in PyQt5 only, (and Spyder-4 still a problem)
  
Areas of particular interest for testers:
- pertinance of ASGI choices

**Comment by stonebig** [comment]
beta1 (plan)

Changes from WinPython 2019-03:
- Infrastructure: 
   - Pythons-3.8.0, Python-3.7.5
   - First WinPython-3.8.0 wheels, classified as final before this cycke
- New Packages: 
   - FastAPI, uvicorn, Quart
   - datasette, python_language_server
- Upgraded Packages: 
- variation per version, for build1:
  -  Spyder-3 is in PyQt5 only, (and Spyder-4 still a problem)
  
beta1 (plan)

Areas of particular interest for testers:
- pertinance/missings in ASGI stack

**Comment by stonebig** [comment]
build1 (2019-11-1X, Python-3.8.0)

Changes from WinPython 2019-04 b1:
- New Packages in Python-3.8: 
   - geopandas, paramiko, pyinstaller, 
- Upgraded Packages: 
   - bokeh-1.4.0, jupyterlab-1.2.3,
   - vscode-1.40.1, PyQt5-5.13.2,
   - still no spyder-4.0.0rc  working,
   - hard fix for "%matplotlib widget"
  
Areas of particular interest for testers:
- python-vscode when upgraded nov. 20 ?

Next build effort:
- python-vscode nov 2019 compability check,
- yet another look for Spyder-4rc2 (currently a problem: hangs at start)

**Comment by stonebig** [comment]
rc (2019-12-22)

Changes from WinPython 2019-04 b2:
- Infrastructure:
 - Python-3.8.1 (with asyncio fixes)
 - Python-3.7.6
- New Packages in Python-3.8: 
   - PySide2-5.14
- Upgraded Packages: 
   - ..., 
   - vscode-1.41.0 of december
  
Areas of particular interest for testers:
- scipy-1.4.0rc1
- python-vscode-2019.11.50794 ... better, but ipywidgets compatibility is now february 2020
 

Next build effort:
- yet another try with Spyder-4..1 ? (when stdio optoin available)

**Comment by stonebig** [comment]
Published ! Now let see if droping Python-2.7 and 32 bit will accelerate the much needed renovation in Python space, around multi-core, garbage collection, and WASM.

**Comment by stonebig** [comment]
Just noticing one binary, winpython-3.7.6.1.exe, didn't fully upload on sourceforge.

It was ok on github though, will try to correct sourceforge soon.

**Comment by stonebig** [comment]
uploade rectified, Winpython64-3.7.6.0.exe is now currectly uploaded also on sourceforge.

 MD5                             | SHA-1                                    | SHA-256                                                          | Binary                          | Size              
---------------------------------|------------------------------------------|------------------------------------------------------------------|---------------------------------|-------------------
788a41c4afb1a17576291de3f30dfe2d | 917a1e8bdb1ef080ca916d3baca83682649f3009 | e523d55a4bbe85d55f45958b7ea5078d56a9aee6772687bee68ba223ad3f7fdd | Winpython64-3.7.6.0.exe         |  679 764 546 Bytes
